src/coreclr/src/vm/eventstore.cpp

```diff
@@ -34,11 +34,9 @@ class EventStore
         EventStoreElem *walk;
         EventStoreElem *next;
 
-        walk = m_Store;
-        while (walk) {
+        for (walk = m_Store; walk; walk = next) {
```

Comment:
This is not a canonical linked-list for-loop. I believe that the original code was easier to understand.

(The other ones are fine.)

Reply:
How is this not canonical?

Reply:
`next` is computed in the middle of the body.